diff -ru gdb-6.6/gdb/config.in gdb-6.6+unixsockets/gdb/config.in
--- gdb-6.6/gdb/config.in	2006-08-08 16:32:15.000000000 -0400
+++ gdb-6.6+unixsockets/gdb/config.in	2008-06-27 18:23:51.000000000 -0400
@@ -67,6 +67,9 @@
 /* Define to 1 if you have the <curses.h> header file. */
 #undef HAVE_CURSES_H
 
+/* Define to 1 if you have the <sys/un.h> header file. */
+#undef HAVE_SYS_UN_H
+
 /* Define to 1 if you have the declaration of `free', and to 0 if you don't.
    */
 #undef HAVE_DECL_FREE
@@ -338,6 +341,9 @@
 /* Define to 1 if `st_blocks' is member of `struct stat'. */
 #undef HAVE_STRUCT_STAT_ST_BLOCKS
 
+/* Define to 1 if `sun_len' is member of `struct sockaddr_un'. */
+#undef HAVE_STRUCT_SOCKADDR_UN_SUN_LEN
+
 /* Define to 1 if `td_pcb' is member of `struct thread'. */
 #undef HAVE_STRUCT_THREAD_TD_PCB
 
diff -ru gdb-6.6/gdb/configure.ac gdb-6.6+unixsockets/gdb/configure.ac
--- gdb-6.6/gdb/configure.ac	2006-12-17 10:38:59.000000000 -0500
+++ gdb-6.6+unixsockets/gdb/configure.ac	2008-06-27 18:23:43.000000000 -0400
@@ -366,6 +366,7 @@
 AC_CHECK_HEADERS(sys/select.h)
 AC_CHECK_HEADERS(sys/syscall.h)
 AC_CHECK_HEADERS(sys/types.h)
+AC_CHECK_HEADERS(sys/un.h)
 AC_CHECK_HEADERS(sys/user.h, [], [],
 [#if HAVE_SYS_PARAM_H
 # include <sys/param.h>
@@ -415,6 +416,7 @@
 
 AC_CHECK_MEMBERS([struct stat.st_blocks])
 AC_CHECK_MEMBERS([struct stat.st_blksize])
+AC_CHECK_MEMBERS([struct sockaddr_un.sun_len],,,[#include <sys/un.h>])
 
 # ------------------ #
 # Checks for types.  #
diff -ru gdb-6.6/gdb/ser-tcp.c gdb-6.6+unixsockets/gdb/ser-tcp.c
--- gdb-6.6/gdb/ser-tcp.c	2006-02-10 17:01:43.000000000 -0500
+++ gdb-6.6+unixsockets/gdb/ser-tcp.c	2008-06-27 18:23:43.000000000 -0400
@@ -1,4 +1,4 @@
-/* Serial interface for raw TCP connections on Un*x like systems.
+/* Serial interface for raw TCP and AF_LOCAL connections on Un*x like systems.
 
    Copyright (C) 1992, 1993, 1994, 1995, 1996, 1998, 1999, 2001, 2005, 2006
    Free Software Foundation, Inc.
@@ -49,6 +49,17 @@
 #include <netinet/tcp.h>
 #endif
 
+#ifdef HAVE_SYS_UN_H
+#define _GNU_SOURCE
+#define _XOPEN_SOURCE_EXTENDED	/* Apparently required on DEC/OSF. */
+#include <sys/un.h>		/* For struct sockaddr_un. */
+#ifndef SUN_LEN
+/* some systems (e.g., Solaris) don't define this handy macro */
+#define SUN_LEN(sau) \
+  (sizeof(*(sau)) - sizeof((sau)->sun_path) + strlen((sau)->sun_path))
+#endif
+#endif /* HAVE_SYS_UN_H */
+
 #include <signal.h>
 #include "gdb_string.h"
 
@@ -70,65 +81,116 @@
 {
   char *port_str, hostname[100];
   int n, port, tmp;
-  int use_udp;
+  enum { USE_TCP, USE_UDP, USE_LOCAL } mode;
   struct hostent *hostent;
-  struct sockaddr_in sockaddr;
+  struct sockaddr_in sockaddri;
+  struct sockaddr *sockaddrp;
+  socklen_t socklen;
+#ifdef HAVE_SYS_UN_H
+  struct sockaddr_un sockaddru;
+#endif
 #ifdef USE_WIN32API
   u_long ioarg;
 #else
   int ioarg;
 #endif
 
-  use_udp = 0;
+  mode =  USE_TCP;
   if (strncmp (name, "udp:", 4) == 0)
     {
-      use_udp = 1;
+      mode = USE_UDP;
       name = name + 4;
     }
   else if (strncmp (name, "tcp:", 4) == 0)
     name = name + 4;
+  else if (strncmp (name, "unix:", 5) == 0)
+    {
+      mode = USE_LOCAL;
+      name = name + 5;
+    }
+  else if (strncmp (name, "local:", 6) == 0)
+    {
+      mode = USE_LOCAL;
+      name = name + 6;
+    }
 
-  port_str = strchr (name, ':');
-
-  if (!port_str)
-    error (_("net_open: No colon in host name!"));	   /* Shouldn't ever happen */
+  if (mode == USE_TCP || mode == USE_UDP)
+    {
+      port_str = strchr (name, ':');
+      port_str = strchr (name, ':');
 
-  tmp = min (port_str - name, (int) sizeof hostname - 1);
-  strncpy (hostname, name, tmp);	/* Don't want colon */
-  hostname[tmp] = '\000';	/* Tie off host name */
-  port = atoi (port_str + 1);
-
-  /* default hostname is localhost */
-  if (!hostname[0])
-    strcpy (hostname, "localhost");
+      if (!port_str)
+        error (_("net_open: No colon in host name!"));	   /* Shouldn't ever happen */
 
-  hostent = gethostbyname (hostname);
-  if (!hostent)
+      tmp = min (port_str - name, (int) sizeof hostname - 1);
+      strncpy (hostname, name, tmp);	/* Don't want colon */
+      hostname[tmp] = '\000';	/* Tie off host name */
+      port = atoi (port_str + 1);
+
+      /* default hostname is localhost */
+      if (!hostname[0])
+        strcpy (hostname, "localhost");
+
+      hostent = gethostbyname (hostname);
+      if (!hostent)
+        {
+          fprintf_unfiltered (gdb_stderr, "%s: unknown host\n", hostname);
+          errno = ENOENT;
+          return -1;
+        }
+      sockaddri.sin_family = AF_INET;
+      sockaddri.sin_port = htons (port);
+      memcpy (&sockaddri.sin_addr.s_addr, hostent->h_addr,
+	      sizeof (struct in_addr));
+      socklen = sizeof(struct sockaddr_in);
+      sockaddrp = (struct sockaddr *)&sockaddri;
+    }
+  else if (mode == USE_LOCAL)
     {
-      fprintf_unfiltered (gdb_stderr, "%s: unknown host\n", hostname);
-      errno = ENOENT;
+#ifdef HAVE_SYS_UN_H
+      if (strlen(name) >= sizeof(sockaddru.sun_path))
+	{
+	  fprintf_unfiltered (gdb_stderr, "%s: name too long\n", name);
+	  errno = EINVAL;
+	  return -1;
+	}
+
+      sockaddru.sun_family = AF_UNIX;
+      strcpy(sockaddru.sun_path, name);
+      socklen = SUN_LEN(&sockaddru);
+#ifdef HAVE_STRUCT_SOCKADDR_UN_SUN_LEN
+      sockaddru.sun_len = socklen;
+#endif
+      sockaddrp = (struct sockaddr *)&sockaddru;
+#else
+      fprintf_unfiltered (gdb_stderr, "%s: socket type not supported\n", name);
+      errno = EINVAL;
       return -1;
+#endif /* HAVE_SYS_UN_H */
     }
+  else
+    error ("net_open: invalid mode!");
 
-  if (use_udp)
+  switch (mode) {
+  case USE_UDP: 
     scb->fd = socket (PF_INET, SOCK_DGRAM, 0);
-  else
+    break;
+  case USE_TCP:
     scb->fd = socket (PF_INET, SOCK_STREAM, 0);
-
+    break;
+  case USE_LOCAL:
+    scb->fd = socket (PF_UNIX, SOCK_STREAM, 0);
+    break;
+  }
   if (scb->fd < 0)
     return -1;
   
-  sockaddr.sin_family = PF_INET;
-  sockaddr.sin_port = htons (port);
-  memcpy (&sockaddr.sin_addr.s_addr, hostent->h_addr,
-	  sizeof (struct in_addr));
-
   /* set socket nonblocking */
   ioarg = 1;
   ioctl (scb->fd, FIONBIO, &ioarg);
 
   /* Use Non-blocking connect.  connect() will return 0 if connected already. */
-  n = connect (scb->fd, (struct sockaddr *) &sockaddr, sizeof (sockaddr));
+  n = connect (scb->fd, sockaddrp, socklen);
 
   if (n < 0
 #ifdef USE_WIN32API
@@ -220,7 +282,7 @@
   ioarg = 0;
   ioctl (scb->fd, FIONBIO, &ioarg);
 
-  if (use_udp == 0)
+  if (mode == USE_TCP)
     {
       /* Disable Nagle algorithm. Needed in some cases. */
       tmp = 1;

